src/webui/www/private/scripts/client.js

```diff
@@ -1655,7 +1657,15 @@ window.addEventListener("DOMContentLoaded", () => {
 window.addEventListener("load", () => {
     // fetch various data and store it in memory
     window.qBittorrent.Cache.buildInfo.init();
-    window.qBittorrent.Cache.preferences.init();
+    window.qBittorrent.Cache.preferences.init({
+        onSuccess: (pref, responseText) => {
+            if (!pref.dht) {
+                const dhtElement = document.getElementById("DHTNodes");
+                dhtElement.classList.add("invisible");
+                dhtElement.previousElementSibling.classList.add("invisible");
```

Comment:
I don't think it should be done here. The user may switch off the DHT anytime while running.
If possible, it should be done in `processServerState()`.

Reply:
I think it's fine to do it here because changing settings through the WebUI causes refresh so it should be always applied properly, no?? (I tested it). The only case that's not covered is when we change the setting through the GUI while WebUI is opened but I'm not really sure if such situation is worth supporting really.

Reply:
>The only case that's not covered is when we change the setting through the GUI while WebUI is opened but I'm not really sure if such situation is worth supporting really.

The reloading of webui on preference save should not be a hard requirement. I mean we might revise that behavior later.